core/src/main/java/nl/nn/adapterframework/jdbc/JdbcQuerySenderBase.java

```diff
@@ -153,17 +156,37 @@ private PreparedStatement prepareQueryWithColunmsReturned(Connection con, String
 		return con.prepareStatement(query,columnsReturned);
 	}
 
-	protected PreparedStatement prepareQuery(Connection con, String query, boolean updateable) throws SQLException {
+	protected PreparedStatement prepareQuery(Connection con, String query, boolean updateable) throws SQLException, JdbcException {
+		String convertedQuery = convertQuery(con, query.trim());
 		if (log.isDebugEnabled()) {
-			log.debug(getLogPrefix() +"preparing statement for query ["+query+"]");
+			log.debug(getLogPrefix() +"preparing statement for query ["+convertedQuery+"]");
 		}
 		String[] columnsReturned = getColumnsReturnedList();
 		if (columnsReturned!=null) {
-			return prepareQueryWithColunmsReturned(con,query,columnsReturned);
+			return prepareQueryWithColunmsReturned(con,convertedQuery,columnsReturned);
 		}
-		return con.prepareStatement(query,ResultSet.TYPE_FORWARD_ONLY,updateable?ResultSet.CONCUR_UPDATABLE:ResultSet.CONCUR_READ_ONLY);
+		return con.prepareStatement(convertedQuery,ResultSet.TYPE_FORWARD_ONLY,updateable?ResultSet.CONCUR_UPDATABLE:ResultSet.CONCUR_READ_ONLY);
 	}
 
+	protected String convertQuery(Connection connection, String query) throws JdbcException, SQLException {
```

Comment:
Ik zou deze convertQuery() ook verplaatsen naar dbmsSupport

Reply:
in dbmsSupport zit ook de convertQuery. Deze (in JdbcQuerySenderBase) checkt of er geconverteerd moeten worden en splits eventueel een query op in subqueries. Het echte werk verder zit in de convertQuery van dbmsSupport.